compiler/ast/src/statements/iteration.rs

```diff
@@ -29,8 +31,14 @@ pub struct IterationStatement {
     pub type_: Type,
     /// The start of the iteration.
     pub start: Expression,
+    /// The concrete value of `start`.
+    #[serde(skip)]
+    pub start_value: RefCell<Option<Value>>,
     /// The end of the iteration, possibly `inclusive`.
     pub stop: Expression,
+    /// The concrete value of `stop`.
+    #[serde(skip)]
+    pub stop_value: RefCell<Option<Value>>,
```

Comment:
I agree.
With our current design, it would be more natural to get these values on the fly via constant propagation/folding.
However, one of our design requirements is that passes (like const prop and loop unrolling) can be toggled on and off.
Type checking will populate `start_value` and `stop_value` if `start` and `stop` are `Literals`. 
If a user disables constant propagation, we can still perform loop unrolling when the bounds are literals.
If a user enables constant propagation, the const prop pass can compute and store the values in `start_value` and `stop_value` before the loop unrolling pass

TLDR. I went with this approach as it appears to be the simplest way to decouple passes and make them toggle-able. I think we should consider a different design in the future.